### PR TITLE
SDL1: Audio and video support

### LINKED ISSUE #263: SDL1 Audio support
SDL1 doesn't have `SDL_QueueAudio`, which is why there is no audio at all when built with `USE_SDL1`.

In SDL1 done, playback is done by using the callback function. I'm not yet entirely sure how to implement this in SDL1.

### PR DESCRIPTION
**Pull request by glebm**
1. Implements audio support for SDL1 (via a callback)
2. Implements blit scaling and correct video rendering for SDL1 (backport of `SDL_BlitScaled` plus copying the palette on every change).

Fixes #263 

Audio plays clear, no clicking / noise.

### CODE REVIEW

File: SourceS/sdl2_to_1_2_backports.h [hunk 1]
```diff
@@ -244,6 +245,18 @@ inline void SDLBackport_PixelformatToMask(int pixelformat, Uint32 *flags, Uint32
 	}
 }
 
+// A limited implementation of `a.format` == `b.format` from SDL2.
+inline bool SDLBackport_PixelFormatFormatEq(const SDL_PixelFormat *a, const SDL_PixelFormat *b)
+{
+	return a->BitsPerPixel == b->BitsPerPixel && (a->palette != nullptr) == (b->palette != nullptr);
```
Comment (AJenbo) [inline SourceS/sdl2_to_1_2_backports.h]: why the parentheses?

File: SourceS/sdl2_to_1_2_backports.h [hunk 2]
```diff
@@ -244,6 +245,18 @@ inline void SDLBackport_PixelformatToMask(int pixelformat, Uint32 *flags, Uint32
 	}
 }
 
+// A limited implementation of `a.format` == `b.format` from SDL2.
```
Comment (AJenbo) [inline SourceS/sdl2_to_1_2_backports.h]: DocBlock comments are preferred for documenting functions

```suggestion
/**
 * A limited implementation of `a.format` == `b.format` from SDL2.
 */
```

File: SourceS/sdl2_to_1_2_backports.h [hunk 3]
```diff
@@ -264,6 +277,334 @@ SDL_CreateRGBSurfaceWithFormatFrom(void *pixels, Uint32 flags, int width, int he
 	return SDL_CreateRGBSurfaceFrom(pixels, flags, width, height, depth, rmask, gmask, bmask, amask);
 }
 
+//= BlitScaled backport from SDL 2.0.9.
+
+#define SDL_BlitScaled SDL_UpperBlitScaled
```
Comment (AJenbo) [inline SourceS/sdl2_to_1_2_backports.h]: nice work, but did you consider resolution switching instead? It probably will perform better on most devices that supports it.

File: SourceS/sdl2_to_1_2_backports.h (hunk 1 above)
Comment (glebm) [reply SourceS/sdl2_to_1_2_backports.h]: Just for clarity: We're checking that either they both have a palette, or neither of them do (presence of a palette in SDL1 corresponds to the `INDEXED` family of formats in SDL2 as far as I can tell)

File: SourceS/sdl2_to_1_2_backports.h (hunk 2 above)
Comment (glebm) [reply SourceS/sdl2_to_1_2_backports.h]: Done

File: SourceS/sdl2_to_1_2_backports.h (hunk 3 above)
Comment (glebm) [reply SourceS/sdl2_to_1_2_backports.h]: I haven't thought of that. It'ss already playing at full speed on my fairly underpowered device, so I've left a note for now.

File: SourceS/sdl2_to_1_2_backports.h (hunk 1 above)
Comment (AJenbo) [reply SourceS/sdl2_to_1_2_backports.h]: Oh I missed that you where comparing the two bools to each other afterwards.

**Comment by glebm** [comment]
> I'm not sure I can maintain it so I hope your up for that, probably not much will affect it though as the video playback is already pretty solid.

I've copied the complicated functions like Blit directly from SDL2 with just enough changes to make it work on SDL1. I'm hoping maintaining this will be a non-issue! Always happy to step in if major changes are happening, I think I have a decent enough grasp of SDL 1 and 2 by now.

File: SourceS/sdl2_to_1_2_backports.h (hunk 3 above)
Comment (AJenbo) [reply SourceS/sdl2_to_1_2_backports.h]: Ok, it probably is a benefit to have both options available, some devices might not change resolution in a smooth way and others may need the reduced load.

**Comment by AJenbo** [comment]
Sounds good. It defiantly would be a big help if you could look at solving the SDL1 build issues for the switch-next branch :D

**Pull request merged**